### PR TITLE
Update AAVE Emissions in StakedTokens

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Add payload to update emissions.
Add tests.

### CODE REVIEW

**Comment by sakulstra** [comment]
aip missing, converted to draft in the meantime

File: src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    IStakeToken.AssetConfigInput[] memory abptConfigs = new IStakeToken.AssetConfigInput[](1);
+    abptConfigs[0] = IStakeToken.AssetConfigInput({
+      emissionPerSecond: AAVE_EMISSION_PER_SECOND_STK_ABPT,
+      totalStaked: 0, // it's overwritten internally
+      underlyingAsset: AaveSafetyModule.STK_AAVE_WSTETH_BPTV2
+    });
+    IStakeToken(AaveSafetyModule.STK_AAVE_WSTETH_BPTV2).configureAssets(abptConfigs);
+
+    // Allowance for stkGHO to pull funds
+    MiscEthereum.AAVE_ECOSYSTEM_RESERVE_CONTROLLER.approve(
```
Comment (rustboyar) [inline src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol:51]: should it be there?

Comment (efecarranza) [reply src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol:51]: @rustboyar the reserve uses SafeERC20 so it reverts if you're going from a non-zero approve to a non-zero approve.

**Review by rustboyar — APPROVED** [review]

File: src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      0
+    );
+
+    uint256 distributionDuration = IStakeToken(AaveSafetyModule.STK_GHO).distributionEnd() -
+      block.timestamp;
+    MiscEthereum.AAVE_ECOSYSTEM_RESERVE_CONTROLLER.approve(
+      MiscEthereum.ECOSYSTEM_RESERVE,
+      AaveV3EthereumAssets.AAVE_UNDERLYING,
+      AaveSafetyModule.STK_GHO,
+      AAVE_EMISSION_PER_SECOND_STK_GHO * distributionDuration
```
Comment (sakulstra) [inline src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol:64]: hmm, i think this math is wrong.

What about the pending claims until now?
Assuming the approval was correct before, the correct way would essentially be:
`currentAllowance - distributionDuration*previousEmission + distributionDuration*newEmission`or
`currentAllowance + distributionDuration *(newEmission - previousEmission)` which is the same.

Comment (sakulstra) [reply src/20240229_AaveV3Ethereum_AmendSafetyModuleEmissions/AaveV3Ethereum_AmendSafetyModuleEmissions_20240229.sol:64]: alternatively could also be fine to do the math offchain as due to proposal delay would be slightly overestimated.

**Review by sakulstra — APPROVED** [review]

**Review by rustboyar — APPROVED** [review]

**Review by sakulstra — APPROVED** [review]

**Pull request merged**